File: TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */


var twoSum = function(nums, target) {
    for (var i = 0; i <= nums.length-1; i++) {
        for (var j = i + 1; j <= nums.length; j++) {
            if (nums[i]+nums[j] == target) {
                return [i, j];
            }
        }
    }
}


function twoSum_Hash(nums, target) {
    var hashTable = new Object;
    for (var i = 0; i <= nums.length-1; i++) {
        hashTable[i] = nums[i];
    }
}


twoSum([[2,7,11,15]], 9);

twoSum_Hash([[3,2,4]], 5);

File: TwoSum.cpp
using namespace std::vector;

class TwoSum{
public:
    vector<int> twoSumFunc(std::vector<int>& nums, int target) {
        int narray[2];
        
        for (int i = 0; i <= nums.size()-1; i++) {
            for(int j = i + 1; j <=  nums.size(); j++) {
                if (nums[i] + nums[j] == target && i != nums.size()-1) {
                    return {i, j};
                }
            };
        };
    return {};
    }
};


class TwoSum_Hash {
    public:
        vector<int> twoSum(std::vector<int>& nums, int target) {
        int n = nums.size();

        for (int i = 0; i <= nums.size()-1; i++) {
            int hashTable[nums[i]] = i;
        };

        struct Hash_items {
            int* key;
            char* value;
        };

    return {};
    }
};



int main() {

    TwoSum twoSum;

    twoSum.twoSumFunc([4,6,4,7], 10);

};

